torch_sparse/storage.py

```diff
@@ -37,18 +39,34 @@ def __init__(self, row, col, value=None, sparse_size=None, rowptr=None,
             ones = torch.ones_like(row)
             out_deg = segment_add(ones, row, dim=0, dim_size=sparse_size[0])
             rowptr = torch.cat([row.new_zeros(1), out_deg.cumsum(0)], dim=0)
+        else:
+            assert rowptr.dtype == torch.long and rowptr.device == row.device
+            assert rowptr.dim() == 1 and rowptr.size(0) == sparse_size[0] - 1
 
         if colptr is None:
             ones = torch.ones_like(col) if ones is None else ones
             in_deg = scatter_add(ones, col, dim=0, dim_size=sparse_size[1])
             colptr = torch.cat([col.new_zeros(1), in_deg.cumsum(0)], dim=0)
+        else:
+            assert colptr.dtype == torch.long and colptr.device == col.device
+            assert colptr.dim() == 1 and colptr.size(0) == sparse_size[1] - 1
 
         if arg_csr_to_csc is None:
             idx = sparse_size[0] * col + row
```

Comment:
This array is already created beforehand, and argsorted if necessary. Could be interesting to reuse it if possible